Development

akjain96 opened the pull request:
Changes are done, want a review of code.

akjain96 committed:
pushed inventory and ansible playbooks

akjain96 committed:
Added JenkinsFile (pipeline code)

akjain96 committed:
Deleting : No need here.

ihimanshubhutani commented on gitpull.yml:
```diff
@@ -0,0 +1,25 @@
+- name: Pulling git code to ansible slave node
+  hosts: anotherinstance18.04
+  remote_user: ubuntu
+  sudo: yes
```
Should you use sudo here?

akjain96 replied:
In code (gitpull.yml), when doing git pull it creates problem, Error : error: cannot open .git/FETCH_HEAD: Permission denied", "stderr_lines": ["error: cannot open .git/FETCH_HEAD: Permission denied"], thats why used sudo: yes. 

We can also use become:yes instead.

ihimanshubhutani commented on installation.yml:
```diff
@@ -0,0 +1,81 @@
+- name: Update ansible slave node with python and pip
+  hosts: anotherinstance18.04
+  remote_user: ubuntu
+  become: yes
```
if you are already using **sudo**, then why you used **become** here or you just copy-pasted my given code here without knowing the underlying thing why I have done such stuff?

akjain96 replied:
As I saw, become is an advanced version of sudo, and is required in the code. As some packages do require that. Please suggest accordingly. 

Example : fatal: [pythoninstanew]: FAILED! => {"changed": false, "msg": "Failed to lock apt for exclusive operation"}

ihimanshubhutani commented on installation.yml:
```diff
@@ -0,0 +1,81 @@
+      state: started
+      enabled: yes
+
+  - name: Test Docker with hello world example
```
Why you need to run Docker testing here, Is that we should do it on the production level?

ihimanshubhutani commented on installation.yml:
```diff
@@ -0,0 +1,81 @@
+      dest: /usr/local/bin/docker-compose
+      mode: 'u+x,g+x'
+      
+  - name: Reboot server in order for changes to take place
```
Are you sure you should restart **SERVER** ?? doesn't it cause downtime to my app?

ihimanshubhutani commented on pydockerimagebuild.yml:
```diff
@@ -0,0 +1,11 @@
+- name: Making Docker Image from DockerFile
+  hosts: anotherinstance18.04
+  remote_user: ubuntu
+  sudo: yes
```
Again sudo?

akjain96 replied:
docker-compose require permissions as sudo to execute. But, added ubuntu user to docker group. so, needed sudo.

ihimanshubhutani commented on dev.inv:
```diff
@@ -0,0 +1,5 @@
+[anotherinstance18.04]
+3.109.53.224 ansible_user=ubuntu ansible_ssh_private_key_file='/var/lib/jenkins/testAnsible.pem'
```
why to use pem file name as camelcase, is that we follow as name convention?

akjain96 replied:
Will make sure from next time that I will not use camel casing format

ihimanshubhutani commented on gitpull.yml:
```diff
@@ -0,0 +1,25 @@
+    when: not (CRUDApp.stat.exists and CRUDApp.stat.isdir)
+  
+  - name: Fetching latest changes
+    command: git fetch --all
```
why you are using fetching and pulling all together, Does it really makes sense?

ihimanshubhutani commented on installation.yml:
```diff
@@ -0,0 +1,81 @@
+    shell: "docker run hello-world"
+    register: hello_world_output
+
+  - name: Show output of hello world example
```
Is it really necessary to run your flask app?

akjain96 replied:
Hi himanshu, this is not our flask app, this is just to make sure that docker is working on VM where we are deploying packages. Please suggest accordingly.

ihimanshubhutani commented on pydockerimagebuild.yml:
```diff
@@ -0,0 +1,11 @@
+  tasks:
+  - name: Creating docker containers
+    command: 
+      cmd: docker-compose up --detach
```
should your docker container keep running if I restart the server?

ihimanshubhutani commented on JenkinsFile:
```diff
@@ -0,0 +1,25 @@
+pipeline {
```
_This review comment includes notes on the way you wrote up you commit messages_

- have you checked how commit messages are really written, have you read git commit message semantics?
- do we really write commit messages as past sentences?
- how your first commit on this PR makes sense?

akjain96 replied:
readed git commit message semantics, will work according to that from next time

ihimanshubhutani commented on README.md:
```diff
@@ -1 +0,0 @@
-# DeployPythonApp
```
you already been told that your main branch should only have one commit ie. Initial commit, why you still pushed second commit on it and made changes to README.md?

ihimanshubhutani commented:
@akjain96 This branch has merge conflicting files, How you creating PR with merge conflicts in it ??

akjain96 committed:
done changes: sudo->become, removed line 22

akjain96 committed:
updated code as per PR comments. 

removed reboot server task, testing docker task and taking output of docker task

akjain96 committed:
fixed code: changed sudo with become.

akjain96 committed:
changed code key with testansible.pem

testAnsible.pem -> testansible.pem
